util/resolver.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/naming"
+)
+
+// FixedBackends implements the naming.Watcher interface by returning
+// a fixed set of servers.
+type FixedBackends struct {
+	pendingServers []string
+}
+
+// Next returns a set of updates describing changes to available servers;
+// it will return the original set on first invocation, and block forever
+// thereafter.
+func (fb *FixedBackends) Next() ([]*naming.Update, error) {
+	if len(fb.pendingServers) == 0 {
+		// Block until there is an update.  There will never be an update.
+		select {} // block forever
```

Comment:
Would it make sense to have this hang on an unexported `chan` in the struct which gets closed by `Close()` and then returns an error?